service-base/src/test/java/org/eclipse/hono/service/http/TenantTraceSamplingHandlerTest.java

```diff
@@ -0,0 +1,547 @@
+    @BeforeEach
+    public void setUp() {
+        tenantClient = mock(TenantClient.class);
+        doAnswer(invocation -> {
+            return Future.succeededFuture(TenantObject.from(invocation.getArgument(0), true));
+        }).when(tenantClient).get(anyString(), any(SpanContext.class));
+
+        final TenantClientFactory tenantClientFactory = mock(TenantClientFactory.class);
+        when(tenantClientFactory.connect()).thenReturn(Future.succeededFuture(mock(HonoConnection.class)));
+        doAnswer(invocation -> {
+            final Handler<AsyncResult<Void>> shutdownHandler = invocation.getArgument(0);
+            shutdownHandler.handle(Future.succeededFuture());
+            return null;
+        }).when(tenantClientFactory).disconnect(any(Handler.class));
+        when(tenantClientFactory.getOrCreateTenantClient()).thenReturn(Future.succeededFuture(tenantClient));
+
+        config = new ProtocolAdapterProperties();
+        config.setSingleTenant(false);
+
+        span = mock(Span.class);
+        final SpanContext spanContext = mock(SpanContext.class);
+        when(span.context()).thenReturn(spanContext);
+
+        ctxMap = new HashMap<>();
+        ctxMap.put(TracingHandler.CURRENT_SPAN, span);
+
+        ctx = mock(RoutingContext.class);
+        when(ctx.get(anyString())).thenAnswer(invocation -> {
+            return ctxMap.get(invocation.getArgument(0));
+        });
+
+        final HttpContextTenantAndAuthIdProvider tenantAndAuthIdProvider = new HttpContextTenantAndAuthIdProvider(
+                config, tenantClientFactory, PARAM_TENANT, PARAM_DEVICE_ID);
+        tenantTraceSamplingHandler = new TenantTraceSamplingHandler(tenantAndAuthIdProvider);
+    }
+
+    /**
+     * Verifies that the handler sets the appropriate span tag for a request for a tenant with 'trace-sampling-mode' set
+     * to 'all'.
+     */
+    @Test
+    public void testHandleSetsSamplingPriorityForMatchingTenant() {
+        // GIVEN a tenant where 'trace-sampling-mode' is set to 'all'
+        final String tenantId = "testTenant";
+        final String authId = "testAuthId";
+
+        final TenantObject testTenant = TenantObject.from(tenantId, true);
+        testTenant.setProperty(TenantConstants.FIELD_TRACE_SAMPLING_MODE, "all");
+        doAnswer(invocation -> {
+            return Future.succeededFuture(testTenant);
+        }).when(tenantClient).get(eq(tenantId), any(SpanContext.class));
+
+        // WHEN handling a request for that tenant
+        final String authorization = "BASIC " +
+                Base64.getEncoder().encodeToString((authId + "@" + tenantId + ":password").getBytes(StandardCharsets.UTF_8));
+        final MultiMap headers = mock(MultiMap.class);
+        when(headers.get(eq(HttpHeaders.AUTHORIZATION))).thenReturn(authorization);
+        final HttpServerRequest req = mock(HttpServerRequest.class);
+        when(req.headers()).thenReturn(headers);
+        when(ctx.request()).thenReturn(req);
```

Comment:
there seems to be a lot of redundant code with setting up the fixture among the test methods. Maybe you can factor out some of the mock creation code into a helper method?

Reply:
Right, I've extracted some methods there.
(Also made a small change in `TenantTraceSamplingHelper` in preparation for support in other adapters.)